src/plugins/validation/2and3/semantic-validators/discriminator.js

```diff
@@ -0,0 +1,74 @@
+// Assertation 1:
+// if discriminator exist inside schema object, it must be of type Object
+
+// Assertion 2:
+// discriminator object must have a field name propertyName
+
+// Assertation 2:
+// propertyName is of type string
+
+// Assertation 3:
+// properties inside a schema object must include propertName from discriminator object
+
+const each = require('lodash/each');
+const has = require('lodash/has');
+
+module.exports.validate = function({ jsSpec }) {
+  const errors = [];
+  const warnings = [];
+
+  const schemas = jsSpec.components.schemas;
+
+  const basePath = ['components', 'schemas'];
+
+  each(schemas, (schema, schemaName) => {
+    if (has(schema, 'discriminator')) {
+      const { discriminator } = schema;
+
+      // If discriminator is not an object, error out and return
+      if (typeof discriminator === 'object') {
```

Comment:
In OpenAPI 2.0, the `discriminator` is just a string property containing the name of the discriminator property in the schema.  So what you have done here is correct for OpenAPI 3.0 but not for OpenAPI 2.0.

There are two ways to address this: 1) use alternate paths in this validator for OAS2 vs OAS3, or 2) move this validation to OAS3, and implement a separate (but very similar) validator for OAS2 (aka Swagger).